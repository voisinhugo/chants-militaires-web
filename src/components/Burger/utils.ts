import songs from "../../data";

export const getRandomSong = () =>
  Math.floor(Math.random() * Object.keys(songs).length);
