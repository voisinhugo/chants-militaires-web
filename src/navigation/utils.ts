const BASE_PATH = "/chants-militaires-web";

export const PATHS = {
  song: "/song"
};

export const getPath = (path: string = "") => `${BASE_PATH}${path}`;
