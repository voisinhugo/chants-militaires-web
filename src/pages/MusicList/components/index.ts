export { MusicItem } from "./MusicItem";
export { SearchBar } from "./SearchBar";
