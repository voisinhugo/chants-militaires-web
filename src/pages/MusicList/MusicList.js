import React from "react";
import styled from "styled-components";
import { MusicItem } from "./components";
import songs from "../../data";
import theme from "../../theme";

const sortedSongs = Object.values(songs).sort((a, b) => a.title > b.title);

const Container = styled.div`
  background-color: ${theme.color.background};
`;

export const MusicList = ({ navigation }) => {
  return (
    <Container>
      <ul>
        {sortedSongs.map(item => (
          <MusicItem key={item.id.toString()} item={item} />
        ))}
      </ul>
    </Container>
  );
};
