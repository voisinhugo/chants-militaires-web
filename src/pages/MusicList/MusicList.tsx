import React, { FunctionComponent } from "react";
import styled from "styled-components";
import { MusicItem } from "./components";
import songs from "../../data";
import theme from "../../theme";
import { Header } from "../../components";
import { Song } from "../../react-app-env";
import { HEADER_HEIGHT } from "../../components/Header";

const sortedSongs = Object.values(songs).sort((a: Song, b: Song) =>
  a.title > b.title ? 1 : -1
);

const ListContainer = styled.div`
  background-color: ${theme.color.background};
  padding-top: ${theme.margin.x1 + HEADER_HEIGHT}px;
  padding-bottom: ${theme.margin.x1}px;
`;

export const MusicList: FunctionComponent = () => {
  return (
    <>
      <Header title="Liste des chants" />
      <ListContainer>
        {sortedSongs.map((item: Song) => (
          <MusicItem key={item.id.toString()} song={item} />
        ))}
      </ListContainer>
    </>
  );
};
