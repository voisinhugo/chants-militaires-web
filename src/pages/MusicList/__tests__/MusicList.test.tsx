import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MusicList } from "../MusicList";
import { renderPage } from "../../../tests/utils";

const song0Id = 0;
const song0Title = "toto";
jest.mock("../../../data/index", () => ({
  "0": { id: 0, title: "toto" },
  "1": { id: 1, title: "tata" }
}));

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe("MusicList", () => {
  it("should render", () => {
    const musicList = renderPage(<MusicList />);

    expect(musicList).toMatchSnapshot();
  });
  it("should navigate to lyric page", () => {
    const { getByText } = render(<MusicList />);

    const itemButton = getByText(song0Title);
    fireEvent.click(itemButton);

    const expectedPath = `/chants-militaires-web/song/${song0Id}`;
    expect(mockHistoryPush).toHaveBeenCalledWith(expectedPath);
  });
});
