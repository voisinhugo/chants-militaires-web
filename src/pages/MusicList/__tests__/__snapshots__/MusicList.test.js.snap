// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicItem should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AykKE jxVPSR"
      >
        <h1
          class="sc-AykKF dltmE"
        >
          Liste des chants
        </h1>
      </div>
      <div
        class="sc-AykKG cYNCoE"
      >
        <div
          class="sc-AykKC hvfnPU"
        >
          <p
            class="sc-AykKD cTiopU"
          >
            toto
          </p>
        </div>
        <div
          class="sc-AykKC hvfnPU"
        >
          <p
            class="sc-AykKD cTiopU"
          >
            tata
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKE jxVPSR"
    >
      <h1
        class="sc-AykKF dltmE"
      >
        Liste des chants
      </h1>
    </div>
    <div
      class="sc-AykKG cYNCoE"
    >
      <div
        class="sc-AykKC hvfnPU"
      >
        <p
          class="sc-AykKD cTiopU"
        >
          toto
        </p>
      </div>
      <div
        class="sc-AykKC hvfnPU"
      >
        <p
          class="sc-AykKD cTiopU"
        >
          tata
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
